internal/parsers/markdown.go

```diff
@@ -96,20 +98,30 @@ func ExtractCodeBlocksFromAst(
 			// Extract the code block if it matches the language.
 			case *ast.HTMLBlock:
 				content := extractTextFromMarkdown(&n.BaseBlock, source)
-				match := expectedSimilarityRegex.FindStringSubmatch(content)
+				matches := expectedSimilarityRegex.FindStringSubmatch(content)
 
-				// TODO(vmarcella): Add better error handling for when the
-				// score isn't parsable as a float.
-				if match != nil {
-					score, err := strconv.ParseFloat(match[1], 64)
+				if len(matches) < 3 {
+					break
+				}
+
+				match := matches[1]
+				if match != "" {
+					score, err := strconv.ParseFloat(match, 64)
 					logging.GlobalLogger.Debugf("Simalrity score of %f found", score)
 					if err != nil {
 						return ast.WalkStop, err
 					}
 					lastExpectedSimilarityScore = score
-					nextBlockIsExpectedOutput = true
+				} else {
+					match = matches[2]
+					logging.GlobalLogger.Debugf("Regex %q found", match)
+					if match == "" {
+						return ast.WalkStop, fmt.Errorf("")
+					}
+					lastExpectedRegexString = match
```

Comment:
It would be better to compile the regex immediately after parsing it so that ie can fail fast if the regex isn't valid. I don't think it'd be an ideal experience for the regex instantiation to fail after the scenario execution has begun execution. What're your thoughts on this?